neuralcoref/train/evaluator.py

```diff
@@ -134,7 +133,8 @@ def display_clusters(self, doc_idx=None):
     ########################
     def get_max_score(self, batch, debug=False):
         inputs, mask = batch
-        inputs = tuple(Variable(i, volatile=True) for i in inputs)
+        with torch.no_grad():
+            inputs = tuple(i for i in inputs)
         if self.cuda:
             inputs = tuple(i.cuda() for i in inputs)
```

Comment:
if you do distributed training you will store the current device in a variable and use `tensor.to(device)`
Example of distributed training: https://github.com/huggingface/transformers/blob/master/examples/run_glue.py#L426
How to run a distributed training setup: https://github.com/huggingface/transformers/tree/master/examples#distributed-training
More information on training that are too long and how to solve that: https://medium.com/huggingface/training-larger-batches-practical-tips-on-1-gpu-multi-gpu-distributed-setups-ec88c3e51255